internal/spokes/spokes.go

```diff
@@ -294,16 +317,37 @@ func (r *spokesReceivePack) performReferenceDiscovery(ctx context.Context) error
 		return nil
 	}
 
+	excludeArgv := []string{"for-each-ref", refAdvertisementFmtArg}
+	for _, ref := range hidden {
+		excludeArgv = append(excludeArgv, fmt.Sprintf("--exclude=%s", ref))
+	}
+
 	p := pipe.New(pipe.WithDir("."), pipe.WithStdout(r.output))
 	p.Add(
-		pipe.Command("git", "for-each-ref", refAdvertisementFmtArg),
+		pipe.Command("git", excludeArgv...),
 		pipe.LinewiseFunction(
 			"collect-references",
 			func(ctx context.Context, _ pipe.Env, line []byte, stdout *bufio.Writer) error {
 				return advertiseRef(line)
 			},
 		),
 	)
+
+	if len(unhidden) > 0 {
+		unhiddenArgv := []string{"for-each-ref", refAdvertisementFmtArg}
+		unhiddenArgv = append(unhiddenArgv, unhidden...)
+
+		p.Add(
+			pipe.Command("git", unhiddenArgv...),
+			pipe.LinewiseFunction(
+				"collect-references",
+				func(ctx context.Context, _ pipe.Env, line []byte, stdout *bufio.Writer) error {
+					return advertiseRef(line)
+				},
+			),
+		)
+	}
```

Comment:
Doing this at the end would end up generating a list of refs that's not sorted alphabetically, and one that could have duplicate refs (if there were only rules for unhiding refs and not any for hiding them). The former is likely to happen for internal pushes from svnbridge; the latter is unlikely to happen because of our prod config. Do either of these matter?

Reply:
Good question. I agree that I think the latter is impossible to trigger in our configuration, since we're always unhiding a subset of hidden references. I double checked the relevant `send-pack` code, and we should be fine to either send duplicate or out-of-order references.